dali/base/dasds.cpp

```diff
@@ -6361,46 +6312,112 @@ void CCovenSDSManager::checkEnvironment(IPropertyTree *oldEnv, IPropertyTree *ne
+    try
     {
-    public:
-        CSDSTreeMaker(IPTreeNodeCreator *nodeCreator) : CPTreeMaker(ipt_none, nodeCreator) { }
-        virtual void endNode(const char *tag, unsigned length, const void *value, bool binary, offset_t endOffset)
+        Owned<IPropertyTree> result;
+
+        OwnedIFile iFileStore = createIFile(storeFilename);
+        OwnedIFileIO iFileIOStore = iFileStore->open(IFOread);
+        if (!iFileIOStore)
+            throw MakeSDSException(SDSExcpt_OpenStoreFailed, "%s", storeFilename.str());
+
+        offset_t fSize = iFileIOStore->size();
+        unsigned crc{0};
+        PROGLOG("Loading %s store %u (size=%.2f MB, storedCrc=%x)", isBinary ? "Binary" : "XML", iStoreHelper->queryCurrentEdition(), ((double)fSize) / 0x100000, storedCrc);
+
+        Owned<ISerialInputStream> serialStream = createSerialInputStream(iFileIOStore);
+        Owned<ISerialInputStream> progressStream = createProgressStream(serialStream, 0, fSize, "Load progress", 60);
+        Owned<ICrcSerialInputStream> crcSerialStream = createCrcInputStream(progressStream);
+        Owned<IBufferedSerialInputStream> bufInStream = createBufferedInputStream(crcSerialStream, bufferSize);
+        if (isBinary)
+            result.setown(createPTreeFromBinary(*bufInStream, nodeCreator));
+        else
         {
-            IPropertyTree *node = queryCurrentNode();
-            CPTreeMaker::endNode(tag, length, value, binary, endOffset);
-            if (((CServerRemoteTree *)node)->testExternalCandidate())
-                convertQueue.append(*(CServerRemoteTree *)node);
+            Owned<IPTreeMaker> treeMaker = createPTreeMaker(ipt_none, nullptr, nodeCreator);
+
+            class CSerialInputStreamWrapper : public CSimpleInterfaceOf<ISimpleReadStream>
+            {
+                ISerialInputStream *stream;
+
+            public:
+                CSerialInputStreamWrapper(ISerialInputStream *_stream) : stream(_stream) {}
+                virtual size32_t read(size32_t len, void *data) override
+                {
+                    return stream->read(len, data);
+                }
+            };
+
+            Owned<ISimpleReadStream> wrapper = new CSerialInputStreamWrapper(bufInStream);
+            result.setown(createPTree(*wrapper, ipt_none, ptr_ignoreWhiteSpace, treeMaker));
         }
-        ICopyArrayOf<CServerRemoteTree> convertQueue;
-    } treeMaker(&nodeCreator);
+
+        bufInStream.clear();
+        crc = crcSerialStream->queryCrc();
+
+        if (storedCrc && storedCrc != crc)
+            IWARNLOG("Error processing store %s - CRC ERROR (file size=%" I64F "d, validation crc=%x, calculated crc=%x)", storeFilename.str(), iFileIOStore->size(), storedCrc, crc);
+
+        return static_cast<CServerRemoteTree *>(result.getClear());
+    }
+    catch (IException *e)
+    {
+        if (!abort)
+            IERRLOG(e, StringBuffer().appendf("Exception - loading %s store file : %s", isBinary ? "Binary" : "XML", storeFilename.str()));
```

Comment:
not everywhere outputs "XML" vs "Binary".
Not sure worth it in general, especially if logging the filename that would make it clear whether it's xml or binary anyway.
Removing would tidy up the code a bit.